ofctrl/ofPacket.go

```diff
@@ -25,6 +25,14 @@ type PacketOut struct {
 	ICMPHeader *protocol.ICMP
 
 	ARPHeader *protocol.ARP
+
+	// EthernetPacket provide a convenient way to fill an ethernet packet into
+	// packetOut without manually setting all L2/3/4 header and data.
+	// If EthernetPacket is provided, all L2/3/4 related fields, namely SrcMAC, DstMAC,
+	// IPHeader, IPv6Header, TCPHeader, UDPHeader, ICMPHeader and ARPHeader, will be
+	// ignored. If EthernetPacket is not used (is nil), an ethernet packet will be
```

Comment:
```suggestion
	// ignored. If EthernetPacket is not used (is nil), an Ethernet packet will be
```

Reply:
Maybe rephrase this comments. I'm a little confused about 
```
// If EthernetPacket is provided, all L2/3/4 related fields, namely SrcMAC, DstMAC,
	// IPHeader, IPv6Header, TCPHeader, UDPHeader, ICMPHeader and ARPHeader, will be
	// ignored. If EthernetPacket is not used (is nil), an ethernet packet will be
// generated based on those L2/3/4 related fields.
```

I thought that you wanted to say that: when calling `getEthernetPacket`, if field `EthernetPacket` has been generated, return it directly; otherwise, generate and return it.

Reply:
Rephrased. And move some comments to `getEthernetPacket`. Hope you can take another look.